Sources/Workspace/Workspace.swift

```diff
@@ -1696,12 +1697,18 @@ extension Workspace {
                 targetName: artifact.targetName
             ]
 
-            if let existingArtifact = existingArtifact, case .remote = existingArtifact.source {
-                // If we go from a remote to a local artifact, we can remove the old remote artifact.
-                artifactsToRemove.append(existingArtifact)
+            if artifact.path.extension == "zip" {
```

Comment:
The "zip" string literal is already used a few times through the codebase, would it make sense to introduce a type/instance property (e.g. `supportedArchiveType/Extension`) to `ManifestLoader`, for example (but that would be a public API change)?

Reply:
thats great yes

Reply:
[Done](https://github.com/apple/swift-package-manager/pull/3746/commits/338312aaeb293cd970e8d05f9e2400cb5229ac42), I hope the approach with a protocol extension works.